R/zchunk_L104.bcoc_en_USA_S_T_Y.R

```diff
@@ -32,49 +32,130 @@ module_emissions_L104.bcoc_en_USA_S_T_Y_DISABLED <- function(command, ...) {
 
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
-    IEA_flow_sector <- get_data(all_data, "energy/IEA_flow_sector")
-    IEA_product_fuel <- get_data(all_data, "energy/IEA_product_fuel")
-    bc.oc_tech <- get_data(all_data, "emissions/bc.oc_tech")
+    IEA_flow_sector <- get_data(all_data, "energy/mappings/IEA_flow_sector")
+    IEA_product_fuel <- get_data(all_data, "energy/mappings/IEA_product_fuel")
+    Bond_bc.oc_tech <- get_data(all_data, "emissions/mappings/Bond_bc.oc_tech")
     GCAM_sector_tech <- get_data(all_data, "emissions/mappings/GCAM_sector_tech")
     L101.in_EJ_R_en_Si_F_Yh <- get_data(all_data, "L101.in_EJ_R_en_Si_F_Yh")
     Bond_BC_1990 <- get_data(all_data, "emissions/Bond_BC_1990")
     Bond_OC_1990 <- get_data(all_data, "emissions/Bond_OC_1990")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+	# Select for USA data
+	Bond_BC_1990 %>%
+    select(Technology, USA) %>%
+	  rename(BC_em = USA) -> Bond_BC_USA_1990
+
+	Bond_OC_1990 %>%
+    select(Technology, USA) %>%
+	  rename(OC_em = USA)  -> Bond_OC_USA_1990
+
+	# Combine BC & OC data and convert to Tg
+	Bond_OC_USA_1990 %>%
+	  left_join_error_no_match(Bond_BC_USA_1990, by=("Technology")) %>%
+	  mutate(OC_em = OC_em * CONV_GG_TG)  %>%
+	  mutate(BC_em = BC_em * CONV_GG_TG) ->
+	  Bond_BCOC_USA_1990
+
+	# Aggregate by GCAM sector and technology
+	Bond_BCOC_USA_1990 %>%
+	  rename(BCOC.Technology = Technology) %>%
+	  # Not all items in the BCOC input dataset have GCAM equivalents, so will have some NAs. 
+  	  # This is expected. Use left_join for this reason.
+	  left_join(Bond_bc.oc_tech,by = c("BCOC.Technology")) %>%
+	  group_by(sector, technology) %>%
+	  summarize_if(is.numeric , sum) %>%
```

Comment:
Are you really summarizing multiple numeric columns here? If so, can they be listed explicitly? If not, something like `summarise(value = sum(value)` may be clearer.